### PR TITLE
feat: Add Targets for NET6 and NetStandard 2.1 | Bump packages

### LINKED ISSUE #363: Why no longer support for netstandard and previous .net versions?
Latest lib supports only net7, I wonder why

### PR DESCRIPTION
**Pull request by jafin**
- Add back support for `.net6` (current LTS)  and `.NetStandard 2.1`
- Bump packages
- Add basic build/test workflow.

**NOTE**: efCore package excludes.NetStandard 2.1 as it is not compatible with EFCore 7 assemblies.

Resolves #363

### CODE REVIEW

File: SmartEnum.sln
Comment (jafin) [inline SmartEnum.sln:1]: Non existant file references removed.

**Comment by ardalis** [comment]
What's the reason for the changes to the build yml file?

**Comment by ardalis** [comment]
Hmm, unfortunately getting lots of build errors with this

**Comment by jafin** [comment]
@ardalis wow i forgot about this, yeah looks like its out of sync with main.  Did you cherry pick bits to get it going or did you want me to correct this PR and remove the build yml stuff?

**Comment by ardalis** [comment]
I mostly tried to do it myself, which worked for the base package but I'm still fighting with the related packages. Probably it's not worth trying to recover anything from this PR unless you disagree.

**Comment by jafin** [comment]
Yep, if youve done the hard work, lets can this one.

**Pull request closed without merging**